### DISCUSSION TITLE
Dark mode

### DISCUSSION THREAD

**Discussion by JensPenneman** [twbs/bootstrap#33102] [Ideas]
# Bootstrap dark mode
## The idea:
It would be fantastic if a darkmode would be integrated (and possibly be disabled by the programmer),
[Bootswatch's darkly theme](https://bootswatch.com/darkly/) would be an example for this one.

![Theme image by Bootswatch](https://64.media.tumblr.com/59f50ae902597307aa62d8e5adaac974/tumblr_inline_p8pesuO29Q1qb17pp_500.png)

## Example code:

You can use all the modes in one css file (after compiling the scss for example):

```CSS
@media (prefers-color-scheme: light) { }
```

But it is also possible to add the dark mode as an option in the html:

```HTML
<link href="./css/bootstrap-dark.css" media="(prefers-color-scheme: dark)" rel="stylesheet">
```

I have also tried to integrate a sort of "dark mode" in my site with the bootstrap framework, the domain where I publicly test everything can be seen under the following link:
[KLJ Stekene TESTS](http://kljstekene.jenspenneman.com/)

**Comment by iamjatinchauhan** [comment]
Bootstrap have Color contrast. It utilizes the [YIQ color space](https://en.wikipedia.org/wiki/YIQ) to automatically return a light `(#fff)` or dark `(#111)` contrast color based on the specified base color.
